src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,243 +33,378 @@
+                " is not a valid date."
+            ) from err
+        if pub_date < datetime.datetime(year=2009, month=1, day=1):
+            raise ValueError(
+                "FEWSNET publishes data since 2009, so adjust your pub_year to"
+                " be >=2009, currently {pub_year}"
+            )
+        elif pub_date > datetime.datetime.now():
+            raise ValueError(
+                "There is no data published in the future. The date should "
+                "refer to the start month-year of the Current "
+                "situation period."
+            )
+
+    def _download_country(
+        self,
+        pub_year: int,
+        pub_month: str,
+        clobber: bool,
+    ) -> Path:
+        """
+        Download fewsnet data that covers the iso2 country.
+
+        Returns
+        -------
+        country_data : Path
+            if data found return the output_dir, else return None
+        """
+        url_country_date = _BASE_URL_COUNTRY.format(
+            iso2=self._iso2, YYYY=pub_year, MM=pub_month
+        )
+
+        return self._download(
             url=url_country_date,
-            zip_filename=zip_filename,
-            output_dir=output_dir_country,
+            area=self._iso2,
+            pub_year=pub_year,
+            pub_month=pub_month,
         )
-    else:
-        country_data = output_dir_country
-    return country_data
-
-
-def _download_fewsnet_region(
-    date_pub: date,
-    region_name: str,
-    region_code: str,
-    output_dir: Path,
-    use_cache: bool,
-) -> Optional[Path]:
-    """
-    Download fewsnet data that covers `region_name`.
 
-    Parameters
-    ----------
-    date_pub : date
-        date for which the data should be downloaded
-        only the year and month part are used
-        this commonly refers to the month of the Current Situation period
-    region_name : str
-        name of the region to which the `iso2` belongs,
-        e.g. "east-africa"
-    region_code : str
-        code that refers to the `region_name,
-        e.g. "EA"
-    output_dir : Path
-        path to dir to which the data should be written
-    use_cache : bool
-        if True, don't download if output_dir already exists
-
-    Returns
-    -------
-    region_data : str
-        If region data exists, return the saved dir else return None
-    """
-    url_region_date = _BASE_URL_REGION.format(
-        region_code=region_code.upper(),
-        region_name=region_name,
-        YYYY=date_pub.year,
-        MM=date_pub.month,
-    )
-
-    output_dir_region = (
-        output_dir / f"{region_code.upper()}{date_pub.strftime('%Y%m')}"
-    )
-    zip_filename_region = f"{output_dir_region.name}.zip"
-    if not output_dir_region.exists() or use_cache is False:
-        region_data = _download_zip(
+    def _download_region(
+        self,
+        pub_year: int,
+        pub_month: str,
+        clobber: bool,
```

Comment:
`clobber` isn't propagated through. I'm assuming it's meant to go through to `_download()` and then `_download_zip()` which is decorated?